CONTRIBUTING.md

```diff
@@ -22,21 +22,86 @@ into three categories:
 
 ## Code contributions
 
+### Setting up a development environment
+
+#### Docker development container
+
+For first-time and occasional contributors to cuML, we encourage you to use the
+`quick_dev.sh` script, which will allow you to build and test your changes
+quickly within a Docker container rather than setting up a build environment
+yourself. This require that you have
+[Docker](https://docs.docker.com/get-docker/) and [the NVIDIA container toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker) installed.
+
+Once you have those prerequisites installed, simply clone the cuML repo and run
+the `quick_dev.sh` script from the root of the cloned repository:
+```bash
+git clone git@github.com:rapidsai/cuml.git
+cd cuml
+./quick_dev.sh
+```
+The first time you run this, it may take some time to download the development
+Docker image. The script will then
+[mount](https://docs.docker.com/storage/bind-mounts/) the cuML source directory
+into the container and provide you with a prompt from within the container. You
+can edit source files as you would normally would in the `cuml` directory on
+your machine and then build from within the container by running `./build.sh`
+from the container prompt.
+
+The development container can be further configured with the following
+environment variables, if desired:
+
+* Basic Options
+    - `CUML_OS`: The name of one of the supported operating systems for cuML's
+      development containers (Default: `ubuntu18.04`)
+    - `CUML_CUDA`: The desired CUDA version (Default: `11.0`)
+    - `CUML_PY`: The desired Python version (Default: `3.8`)
+    - `CUDA_VISIBLE_DEVICES`: The GPU(s) that will be visible in the dev
+      container (Default: `all`)
```

Comment:
I would imagine that a significant portion of the target audience for this would do single GPU contributions first, so perhaps I would move this to advanced and add a note along the lines `if you are working on a multigpu contribution or your system has more than one gpu... most users can leave the default value`